wagtail/test/settings.py

```diff
@@ -261,6 +261,12 @@
 WAGTAILREDIRECTS_AUTO_CREATE = False
 
 
+# Disable per-thread caching of site/SiteRootPath data by default, as it tends to
+# cause confusion between tests that modify site data. It can be overridden on a
+# case-by-case basis when needed.
+WAGTAIL_PER_THREAD_SITE_CACHING = False
```

Comment:
I think it's important to document this as something folks might want to set for their test suites to avoid any weirdness. I'm happy to document the setting itself, but it will definitely be worth a mention in the release notes, too.